### ISSUE TITLE
Explicit body parser

### ISSUE THREAD

**Issue by MightyPixel** [balderdashy/sails#2410]
v0.10.5
On a new Sails JS application I uncomment the middleware and bodyParser in config/http.js and lift the server. Accessing the index results a pending request that never ends. Am I doing something wrong?
Sails uses skipper by default, right? Uncommenting this line shouldn't change the default behaviour.

Background: I'm using http-proxy module witch apparently has some issues related to body parsers (https://github.com/nodejitsu/node-http-proxy/issues/180). I get an error when I try to proxy a post request. The server hangs and I get a socket hangup.

If I set  bodyParser: false in config/http.js the post requests that are proxied are fine, but then when I do post request to some of my controllers I get undefined in req.param('some-input-name-field').

So I guess I have two options - A) to use different bodyparser then skipper (That's why I got to this issue) and B) to find out why setting skipper to false results not filling the req.params

Cheers.

**Comment by samtsai** [comment]
Hi guys, I know this might be outdated but @loicsaintroch I see that you closed this issue. Is this still considered a "bug" that Sails team will allow for 'bodyParser' as part of the list of 'middleware' in addition to using a proxy server or is this workaround the only option?

**Comment by moisesfr** [comment]
Hello,

Working with Node.js v4.2.3 and Sails.js v0.11.3, this problem still persists, and the above workaround didn't quite do the trick. 
It freezes the app, the only way to execute the method its including it in the middleware object.

Any ideas on how to solve this? 

Would be much appreciated, thanks in advance.

**Comment by Kiza** [comment]
I am using  Sails v0.12.1.

In the generated ./config/http.js. `bodyParser: require('skipper')` is not at top level as @sgress454 suggested. 


```
module.exports.http = {
  middleware: {
     order: [...all your middleware...], 
     bodyParser: require('skipper')
  },
};
```

It does not work. 

Move bodyParser: require('skipper') to top level resolved it.

```
module.exports.http = {
  middleware: {
     order: [...all your middleware...]
  },
  bodyParser: require('skipper')
}
```

But @sgress454 mentioned it is a deprecated method in 2014. I am not sure whether I am doing it right.

**Comment by sgress454** [comment]
@kiza @moisesfr The bodyparser config issues have been solved in v0.12.2 onward -- see https://github.com/sails101/use-xml-body-parser for an example (specifically https://github.com/sails101/use-xml-body-parser/blob/master/config/http.js).

**Comment by AakashMallik** [comment]
> If you want to supply options for skipper while using the workaround mentioned above, you can do this:
> 
> ```
> bodyParser: function(opts)
> {
>     var opts = _.extend(opts,
>     {
>         verify: function(req,res,body)
>         {
>             console.log(body);
>         }
>     });
>     return require('skipper')(opts);
> }
> ```

Just an FYI, you need to write an Immediately Invoked Function Expression (IIFE) when doing these configurations of middleware as an initialization step. For comparison, you are supposed to give require('skipper')(opt) to bodyParser, not a function which when invoked returns require('skipper')(opt).
